evalml/pipelines/components/transformers/samplers/base_sampler.py

```diff
@@ -59,6 +59,52 @@ def transform(self, X, y=None):
             y = infer_feature_types(y)
         return X, None
 
+    def _convert_dictionary(self, sampling_dict, y):
```

Comment:
Should this method live in the base class? It would be different for undersamplers vs oversampler right?

Reply:
I plan on doing an if-statement in the base class to handle the differences between over and undersamplers! Should be a one-line difference